File: appchat/src/main/java/vnua/edu/appchat/entity/Message.java
package vnua.edu.appchat.entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(columnDefinition = "TEXT")
    private String message;
    @ManyToOne
    @JoinColumn(name = "sender_id")
    private User sender;
    @ManyToOne
    @JoinColumn(name = "receiver_id")
    private User receiver; // người nhận
}

File: appchatclient/chatclientspring/src/model/Message.ts
export class Message {
    id: number;
    senderId: number;
    receiverId: number;
    message: string;
    senderName: string;
    receiverName: string;

    constructor(id: number, senderId: number, receiverId: number, message: string, senderName: string, receiverName: string) {
        this.id = id;
        this.senderId = senderId;
        this.receiverId = receiverId;
        this.message = message;
        this.senderName = senderName;
        this.receiverName = receiverName;
    }
}